src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -874,6 +874,403 @@ string M(object o, bool b)
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                if (c != null)
+                {
+
+                }
+
+                var x2 = c;
+            }
+
+            // out of scope
+            var x3 = c;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestOutOfScope()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                if (c != null)
+                {
+
+                }
+            }
+
+            // out of scope
+            var x3 = c;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestDeclarationOnOuterBlock()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                if (c != null)
+                {
+
+                }
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        {
+            {
+                if (e is C c)
+                {
+
+                }
+            }
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestConditionalExpression()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        M(c != null ? c : null);
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        M(e is C c ? c : null);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestConditionalExpression_OppositeBranch()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e, C x)
+    {
+        [|var|] c = e as C;
+        M(c != null ? c : null, c);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_NoInlineTypeCheck()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e, C x)
+    {
+        [|var|] c = e as C;
+        for (;(c)!=null;) {}
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_InlineTypeCheck()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|C|] c;
+        for (; !((c = e as C)==null);) { }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        for (; !(!(e is C c));) { }
```

Comment:
idea for feature.  reduce double negatives.

Reply:
I suppose that should be a part of `Microsoft.CodeAnalysis.Simplification.Simplifier`

Reply:
👍